### ISSUE TITLE
RhythmS, RTEs and break tags (oh MY!)

### ISSUE THREAD

**Issue by drolsen** [connectivedx/fuzzy-chainsaw#178]
Think I found something of note worth:

https://github.com/connectivedx/fuzzy-chainsaw/blob/develop/source/tags/Rhythm/Rhythm.css
Our Rhythm code uses a * + * pattern

https://github.com/connectivedx/fuzzy-chainsaw/blob/develop/source/tags/RichText/RichText.css
and our RichText tag (something we used for SiteCore RTE hook over various components) extends the Rhythm styles out of the box.

This means if our clients from RTE ever have this instance:
{code}
<p>
PO BOX<br>
1234 First Street<br>
Gladstone Oregon, 97777
</p>
{code}

The second instance of any break tag will get an artificial margin-top: 1em;. Another instance is something like this:
{code}
<p>This is some super fantastic bit of text <a href="/">with link</a> and a real desire<br>
to break text to next line early.</p>
{code}
Again the secon instance of a break tag would get this artifical margin-top: 1em because it qualifies as * + * thanks to the anchor.
I was able to get around this by chaning my projects Rhythm.css to the following:

{code}
.rhythm,
.rhythm--default { & > * + *:not(br) { margin-top: 1em; } }
.rhythm--small { & > * + *:not(br) { margin-top: 0.5em; } }
.rhythm--large { & > * + *:not(br) { margin-top: 2em; } }

.rhythm-deep,
.rhythm-deep--default { & * + *:not(br) { margin-top: 1em; } }
.rhythm-deep--small { & * + *:not(br) { margin-top: 0.5em; } }
.rhythm-deep--large { & * + *:not(br) { margin-top: 2em; } }
{code}

and almost considered adding :not(span) as well as these inline elements should always be ignored by rhythm.

Thanks!

**Comment by krambuhl** [comment]
Oh yea! this will be fun one. It seems reasonable that Rhythm shouldn't provide rhythm to inline elements.  but this seems more directly related to rte than rhythm, I think I would go about it by doing something like this:

```
.rte {
  br { margin-top: 0 }
}
```

That might not work as expected because of how specific the rhythm selector is.  if that's the case, the solution you have above will work, but you'd be making that selector even more specific. That's not the worst, but anytime we want to "undo" the rhythm styles we will have to be more specific or use important.

I'm not sure I have a solid answer for this problem.  This is really only an issue where the rhythm is applied to user entered rte's; If a developer is using rhythm they are probably aware of that limitation.  I wonder if the solution is to define rhythm specific to RTE instead of extending rhythm, that way this solution doesn't fallout to all sections using rhythm.

**Comment by drolsen** [comment]
Yeah, if we try to overload the Rhythm pattern we would be overloading vendor defaults, so I'm not sure (my guess is it will look exactly the same) what that would do in chrome vs ff, but indeed I think would require an !important IIR.

If we did have a RTE only rhythm pattern, we would have 0 overloads for break tags and would be 100% vendor rules (something I'm very fond of).. but I agree, rock and a hard spot.
